pytesmo/validation_framework/adapters.py

```diff
@@ -160,6 +160,68 @@ def read(self, *args, **kwargs):
         data = super(SelfMaskingAdapter, self).read(*args, **kwargs)
         return self.__mask(data)
 
+
+class AdvancedMaskingAdapter(BasicAdapter):
```

Comment:
This is just a strictly more flexible implementation for a `SelfMaskingAdapter` right? Would it make sense to inherit `SelfMaskingAdapter` from this class to show that it basically does the same but with less options? It just might just confuse people which adapter they should use?

Reply:
It fills the same evolutionale niche as the SelfMaskingAdapter, yeah. Personally, I wouldn't change the SelfMaskingAdapter at all (since I only left it in for compatability). Well, maybe change it to add a deprecation warning - and then remove it some time in the future.
If we add a deprecation warning and a comment a la 'please use `AdvancedMaskingAdapter` instead', this should help users decide what to use.